vcd/resource_vcd_org_vdc.go

```diff
@@ -536,6 +600,152 @@ func resourceVcdVdcDelete(d *schema.ResourceData, meta interface{}) error {
+	if vcdClient.Client.APIVCDMaxVersionIs(">= 33.0") {
+		log.Printf("[TRACE] updating assigned VM sizing policies to VDC")
+
+		_, idsOk := d.GetOk("vm_sizing_policy_ids")
+		_, defaultIdOk := d.GetOk("default_vm_sizing_policy_id")
+
+		if idsOk != defaultIdOk {
+			return fmt.Errorf("both fields are required `vm_sizing_policy_ids` and `default_vm_sizing_policy_id`")
+		}
+
+		vcdClient := meta.(*VCDClient)
+		vcdComputePolicyHref, err := vcdClient.Client.OpenApiBuildEndpoint(types.OpenApiPathVersion1_0_0, types.OpenApiEndpointVdcComputePolicies)
+		if err != nil {
+			return fmt.Errorf("error constructing HREF for compute policy")
+		}
+
+		adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
+		if err != nil {
+			return fmt.Errorf(errorRetrievingOrg, err)
+		}
+
+		vdc, err := adminOrg.GetAdminVDCByName(d.Get("name").(string), false)
+		if err != nil {
+			return fmt.Errorf(errorRetrievingVdcFromOrg, d.Get("org").(string), d.Get("name").(string), err)
+		}
+
+		if d.HasChange("default_vm_sizing_policy_id") && !d.HasChange("vm_sizing_policy_ids") {
+			defaultPolicyId := d.Get("default_vm_sizing_policy_id").(string)
+			vdc.AdminVdc.DefaultComputePolicy = &types.Reference{HREF: vcdComputePolicyHref.String() + defaultPolicyId, ID: defaultPolicyId}
+			_, err := vdc.Update()
+			if err != nil {
+				return fmt.Errorf("error setting default VM sizing policy. %s", err)
+			}
+		}
+
+		if d.HasChange("vm_sizing_policy_ids") && !d.HasChange("default_vm_sizing_policy_id") {
+			vmSizingPolicyIdStrings := convertSchemaSetToSliceOfStrings(d.Get("vm_sizing_policy_ids").(*schema.Set))
+			policyReferences := types.VdcComputePolicyReferences{}
+			var vdcComputePolicyReferenceList []*types.Reference
+			for _, policyId := range vmSizingPolicyIdStrings {
+				vdcComputePolicyReferenceList = append(vdcComputePolicyReferenceList, &types.Reference{HREF: vcdComputePolicyHref.String() + policyId})
+			}
+			policyReferences.VdcComputePolicyReference = vdcComputePolicyReferenceList
+
+			_, err = vdc.SetAssignedComputePolicies(policyReferences)
+			if err != nil {
+				return fmt.Errorf("error setting VM sizing policies. %s", err)
+			}
+		}
+
+		err = changeVmSizingPoliciesAndDefaultId(d, vcdComputePolicyHref.String(), vdc)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// changeVmSizingPoliciesAndDefaultId handles VM sizing policies. Created VDC generates default VM sizing policy which requires additional handling.
+// Assigning and setting default VM sizing policies requires different API calls. Default policy can't be removed, as result
+// we approach this with adding new policies, set new default, remove all old policies.
+func changeVmSizingPoliciesAndDefaultId(d *schema.ResourceData, vcdComputePolicyHref string, vdc *govcd.AdminVdc) error {
+	if d.HasChange("default_vm_sizing_policy_id") && d.HasChange("vm_sizing_policy_ids") {
+		vmSizingPolicyIdStrings := convertSchemaSetToSliceOfStrings(d.Get("vm_sizing_policy_ids").(*schema.Set))
```

Comment:
Should we validate `default_vm_sizing_policy_id` does exist in new `vm_sizing_policy_ids` list to throw an error?

Reply:
it's sugar - can add it

Reply:
If the error from API is clear - no need.

Reply:
added